test/integration/utils_test.go

```diff
@@ -53,14 +53,9 @@ func exitOnErrWithCode(ctx context.Context, err error, exitCode int) {
 	}
 
 	fmt.Println("WARNING: failure occurred, performing test cleanup")
-	if env != nil && testenv.ExistingClusterName() == "" && testenv.KeepTestCluster() == "" {
-		ctx, cancel := context.WithTimeout(ctx, environmentCleanupTimeout)
-		defer cancel()
-
-		fmt.Printf("INFO: cluster %s is being deleted\n", env.Cluster().Name())
-		if cleanupErr := env.Cleanup(ctx); cleanupErr != nil {
-			err = fmt.Errorf("cleanup failed after test failure occurred CLEANUP_FAILURE=(%w): %w", cleanupErr, err)
-		}
+	fmt.Printf("INFO: cluster %s is being deleted\n", env.Cluster().Name())
```

Comment:
Wouldn't it be better if we put this in the `RemoveCluster` only when the actual removal takes place? And maybe log that we're skipping removal because reasons?